### ISSUE TITLE
Clean up deployment names in istio/installer

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#15541]
We have some names like pilot11 and citadel11. We should either:

* remove the suffix
* if its needed, (and I am not sure why it is), update it automatically from .Release.Version

@costinm not sure why the suffix was added, do we need it?

**Comment by costinm** [comment]
It was added in an attempt to support same namespace. In particular for citadel - which has to go into istio-system, possibly along a 1.2 install.

It is "11" because it was added while 1.0 was the released version, for the gradual upgrade to 1.1 :-)

**Comment by costinm** [comment]
I am working on a doc and prototyping with the idea we discussed - it is clear from feedback that supporting 'upgrade using a single namespace' is important, and that implies we'll need some suffix. 
But it'll not be '11'.

The main problem is the conflicts on 'cluster wide' resources. If helm tiller is used - it'll refuse to install
if there are resources already installed by a different chart. For kubectl - there is a danger that user deletes the previous deployment taking with it the cluster roles. 

The current solution in installer is to have each 'install' create its own distinct cluster roles, service accounts, etc - and because some service names and service account names are hardcoded ( istio-pilot ) we used different namespaces to make it possible to keep the names. For example the SAN checks for  pilot/galley.
